sklearn/manifold/tests/test_isomap.py

```diff
@@ -14,75 +14,112 @@
 eigen_solvers = ['auto', 'dense', 'arpack']
 path_methods = ['auto', 'FW', 'D']
 
+N_PER_SIDE = 5
+NPTS = N_PER_SIDE ** 2
 
-def test_isomap_simple_grid():
-    # Isomap should preserve distances when all neighbors are used
-    N_per_side = 5
-    Npts = N_per_side ** 2
-    n_neighbors = Npts - 1
 
+def create_sample_data(add_noise=False):
     # grid of equidistant points in 2D, n_components = n_dim
-    X = np.array(list(product(range(N_per_side), repeat=2)))
+    X = np.array(list(product(range(N_PER_SIDE), repeat=2)))
+    if add_noise:
+        # add noise in a third dimension
+        rng = np.random.RandomState(0)
+        noise = 0.1 * rng.randn(NPTS, 1)
+        X = np.concatenate((X, noise), 1)
+    return X
+
+
+@pytest.mark.parametrize(
+    "n_neighbors, radius",
+    [(NPTS-1, None),
+     (None, np.inf)]
+)
+def test_isomap_simple_grid(n_neighbors, radius):
+    # Isomap should preserve distances when all neighbors are used
+    X = create_sample_data(add_noise=False)
 
     # distances from each point to all others
-    G = neighbors.kneighbors_graph(X, n_neighbors,
-                                   mode='distance').toarray()
+    if n_neighbors is not None:
+        G = neighbors.kneighbors_graph(X, n_neighbors,
+                                       mode="distance").toarray()
+    else:
+        G = neighbors.radius_neighbors_graph(X, radius,
+                                             mode="distance").toarray()
 
     for eigen_solver in eigen_solvers:
         for path_method in path_methods:
```

Comment:
The modification of this test can also introduce a parametrization on `eigen_solver` and `path_method`.

Reply:
What do yo think, @MaxwellLZH?

Reply:
What do you think, @MaxwellLZH?